File: php/features.php
<?php
declare(strict_types=1);

// Evervault SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EvervaultFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EvervaultBaseFeature();
            case "test":
                return new EvervaultTestFeature();
            default:
                return new EvervaultBaseFeature();
        }
    }
}

File: py/features.py
# Evervault SDK feature factory

from feature.base_feature import EvervaultBaseFeature
from feature.test_feature import EvervaultTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EvervaultBaseFeature(),
        "test": lambda: EvervaultTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
